### PR TITLE
Issue 831: add tasks to update transition_aged_youth when 14

### LINKED ISSUE #831: casa_case becomes transition_aged_youth when they turn 14
**Description**
On the first day of a `casa_case`'s birth month in the year they turn 14 (`birth_month_year_youth` in the `casa_cases` table), 
`transition_aged_youth` should change from `false` to `true`. 

**Acceptance criteria**
- [ ] Backend: Under the conditions listed above, `transition_aged_youth` automatically becomes `true`
- [ ] Frontend: When `transition_aged_youth` is `true`, 🐛 🦋  appears on the front end with the text `Yes`

Screen Shot 2020-09-30 at 8 06 34 PM (2)

### PR DESCRIPTION
**Pull request by jmkoni**
### What github issue is this PR for, if any?
Resolves #831 

### What changed, and why?
Added a scope to check if youth is 14 and not marked as transitioned and a task to update all youth who match that criteria.

### How will this affect user permissions?
Not a user-facing change.

### How is this tested? (please write tests!) 💖💪
Added a test for the scope change.

### CODE REVIEW

**Comment by jmkoni** [comment]
I am unsure why that test is failing... this change should be pretty isolated.

File: spec/models/casa_case_spec.rb [hunk 1]
```diff
@@ -21,6 +21,27 @@
...
+        birth_month_year_youth: Date.current - 13.years,
+        transition_aged_youth: false)
+      transitioned_14_yo = create(:casa_case,
+        birth_month_year_youth: Date.current - 14.years,
+        transition_aged_youth: true)
+      not_transitioned_14_yo = create(:casa_case,
+        birth_month_year_youth: Date.current - 14.years,
+        transition_aged_youth: false)
+      cases = CasaCase.should_transition
+      aggregate_failures do
```
Comment (littleforest) [inline spec/models/casa_case_spec.rb:36]: cool, did not know about `aggregate_failures` method.

**Review by littleforest — APPROVED** [review]
Looks good to me! Can't see what the failure is, but wondering if you need to set your example cases to something like `Date.current - 14.years - 5.minutes` or something?

**Comment by jmkoni** [comment]
@littleforest  the test failure is not my test 😬 that’s why it’s super odd 😅

**Comment by jmkoni** [comment]
Current failure:

7F43298E-D629-4478-B5EB-CEED0BADB474

**Comment by littleforest** [comment]
Oh, didn't realize I could see the Travis build. I can help look a bit later if you haven't figured it out yet.

**Comment by jmkoni** [comment]
@littleforest  yeah if you have a sec and see something, I’d appreciate it! I can’t look too much more today.

**Comment by jmkoni** [comment]
It sounds like @Garbar might be getting a similar failure on her branch

**Pull request merged**